vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java

```diff
@@ -0,0 +1,187 @@
+     */
+    public void tap(WebElement element, Duration duration)
+    {
+        performAction(b -> b.press(elementOption(element))
+                            .waitAction(waitOptions(duration))
+                            .release());
+    }
+
+    /**
+     * Swipes in <b>swipeDuration</b> direction until one of the following conditions happen:
+     * <ul>
+     * <li>the <b>stopCondition</b> return true result</li>
+     * <li>the end of mobile scroll view is reached</li>
+     * <li>the swipe limit is exceeded</li>
+     * </ul>
+     * @param direction direction to swipe, either <b>UP</b> or <b>DOWN</b>
+     * @param swipeDuration duration between a pointer moves from the start to the end of the swipe coordinates
+     * @param stopCondition condition to stop swiping
+     */
+    public void swipeWhile(SwipeDirection direction, Duration swipeDuration, BooleanSupplier stopCondition)
+    {
+        /*
+         * mobile:scroll
+         * iOS
+         * - scroll to an element works only if the element resides in UP scroll direction
+         * - if the element in DOWN scroll direction, the command still tries to find the element in UP scroll direction
+         * - payload is {"element" : "hash", "toVisible" : true}
+         * Android
+         * - supports only accessibility id, uiautomator strategies
+         * - payload is {"strategy": "accessibility id", "selector": "selector"}
+         * */
+
+        int screenshotHeightToCompare = genericWebDriverManager.getSize().getHeight() / HEIGHT_DIVIDER;
+        int heightOffset = screenshotHeightToCompare * 2;
+
+        Duration stabilizationDuration = mobileAppTestConfiguration.getSwipeStabilizationDuration();
+        int swipeLimit = mobileAppTestConfiguration.getSwipeLimit();
+        BufferedImage previousFrame = null;
+        for (int count = 0; count <= swipeLimit; count++)
+        {
+            swipe(direction, swipeDuration);
+            Sleeper.sleep(stabilizationDuration);
+            if (stopCondition.getAsBoolean())
+            {
+                break;
+            }
+            BufferedImage area = takeScreenshot();
+            BufferedImage currentFrame = area.getSubimage(0, area.getHeight() - heightOffset, area.getWidth(),
+                    screenshotHeightToCompare);
+            if (previousFrame != null && ImageTool.equalImage(currentFrame).matches(previousFrame))
+            {
+                break;
+            }
+            previousFrame = currentFrame;
+            if (count == swipeLimit)
+            {
+                throw new IllegalStateException(
+                        String.format("Swiping is stopped due to exceeded swipe limit '%d'", swipeLimit));
+            }
+        }
+    }
+
+    private BufferedImage takeScreenshot()
+    {
+        try
+        {
+            return screenshotTaker.takeViewportScreenshot();
+        }
+        catch (IOException e)
+        {
+            throw new UncheckedIOException(e);
```

Comment:
I believe there is no much sense to wrap it

Reply:
using images is a temporary solution to determine swipes and it would look weird that the swipe gesture throws IOException because it doesn't imply any IO.

Reply:
ok, I buy it